kotlin-result-coroutines/src/jsTest/kotlin/com.github.michaelbull.result.coroutines/binding/AsyncSuspendableBindingTests.kt

```diff
@@ -0,0 +1,74 @@
+package com.github.michaelbull.result.coroutines.binding
+
+import com.github.michaelbull.result.Err
+import com.github.michaelbull.result.Ok
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.coroutines.runBlockingTest
+import kotlinx.coroutines.async
+import kotlinx.coroutines.delay
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class AsyncSuspendableBindingTest {
+
+    private sealed class BindingError {
+        object BindingErrorA : BindingError()
+        object BindingErrorB : BindingError()
+    }
+
+    @Test
+    fun returnsOkIfAllBindsSuccessful(): dynamic {
+        suspend fun provideX(): Result<Int, BindingError> {
+            delay(100)
+            return Ok(1)
+        }
+
+        suspend fun provideY(): Result<Int, BindingError> {
+            delay(100)
+            return Ok(2)
+        }
+
+        return runBlockingTest {
+            val result = binding<Int, BindingError> {
+                val x = async { provideX().bind() }
+                val y = async { provideY().bind() }
+                x.await() + y.await()
+            }
+            assertTrue(result is Ok)
+            assertEquals(
+                expected = 3,
+                actual = result.value
+            )
+        }
+    }
+
+    @Test
+    fun returnsAnyErrIfBindingFailed() {
+        suspend fun provideX(): Result<Int, BindingError> {
+            delay(1)
+            return Ok(1)
+        }
+
+        suspend fun provideY(): Result<Int, BindingError.BindingErrorA> {
+            delay(2)
+            return Err(BindingError.BindingErrorA)
+        }
+
+        suspend fun provideZ(): Result<Int, BindingError.BindingErrorB> {
+            delay(1)
+            return Err(BindingError.BindingErrorB)
+        }
+
+        runBlockingTest {
```

Comment:
does this not need a `return` as well? would that have affected the results discusssed below?

Reply:
Ah yes, my bad. Should be fixed now.